src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -18,23 +39,28 @@ export const TodoFilter: React.FC = () => {
 
       <p className="control is-expanded has-icons-left has-icons-right">
         <input
+          onChange={(e) => handleQuery(e.target.value)}
           data-cy="searchInput"
           type="text"
           className="input"
+          value={query}
           placeholder="Search..."
         />
         <span className="icon is-left">
           <i className="fas fa-magnifying-glass" />
         </span>
 
-        <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-          {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-          <button
-            data-cy="clearSearchButton"
-            type="button"
-            className="delete"
-          />
-        </span>
+        {query.length > 0 && (
+          <span className="icon is-right" style={{ pointerEvents: 'all' }}>
+            {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
+            <button
+              onClick={() => handleClearQuery()}
+              data-cy="clearSearchButton"
+              type="button"
+              className="delete"
+            />
+          </span>
```

Comment:
this pattern looks little weird to me, usually span is into button, and has some content, imo in this case `clear` would be sufficient 😄